File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/client/filter/BasicAuthClientFilter.kt
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.docs.client.filter

//tag::class[]

import io.micronaut.http.MutableHttpRequest
import io.micronaut.http.annotation.ClientFilter
import io.micronaut.http.annotation.RequestFilter
import jakarta.inject.Singleton

@BasicAuth // <1>
@Singleton // <2>
@ClientFilter
class BasicAuthClientFilter {

    @RequestFilter
    fun doFilter(request: MutableHttpRequest<*>) {
        request.basicAuth("user", "pass")
    }
}
//end::class[]

File: test-suite-python/src/test/python/micronaut/docs/client/filter/BasicAuthClientFilter.py
# tag::class[]
from jakarta.inject import Singleton
from micronaut.http import MutableHttpRequest
from micronaut.http.annotation import ClientFilter, RequestFilter

from .BasicAuth import BasicAuth


@BasicAuth  # <1>
@Singleton  # <2>
@ClientFilter
class BasicAuthClientFilter:

    @RequestFilter
    def filter(self, request: MutableHttpRequest) -> None:
        request.basicAuth("user", "pass")
# end::class[]
